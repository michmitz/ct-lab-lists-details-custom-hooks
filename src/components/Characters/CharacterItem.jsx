import React from 'react';
import PropTypes from 'prop-types';

const CharacterItem = ({ name, imageUrl }) => {
  return <figure>
    <img src={imageUrl} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>
}

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
}

export default CharacterItem