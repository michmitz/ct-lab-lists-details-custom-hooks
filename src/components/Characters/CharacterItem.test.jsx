import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('CharacterItem component', () => {
  afterEach(() => cleanup());

  it('renders single character', () => {
    const { asFragment } = render(<CharacterItem
      name='Michelle'
      imageUrl='url@url.com'
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});