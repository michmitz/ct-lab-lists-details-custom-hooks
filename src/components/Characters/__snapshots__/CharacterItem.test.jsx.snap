// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Michelle"
      src="url@url.com"
    />
    <figcaption>
      Michelle
    </figcaption>
  </figure>
</DocumentFragment>
`;
